### PR TITLE
OBGM-564 items disappearing from pack and send page after editing revised qty

### PR DESCRIPTION
**Pull request by drodzewicz**

### CODE REVIEW

**Comment by drodzewicz** [comment]
While investigating this issue there were many annoying eslint errors in the browser console so I decided to fix some of them.
Most of them were related to the defined proptypes that didn;t match the passed prop data

**Review by kchelstowski — APPROVED** [review]
as for those ESLint issues, we need to adjust settings to what was in Grails 1, because I agree it's annoying

File: grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy [hunk 1]
```diff
@@ -2366,7 +2366,7 @@ class StockMovementService {
     }
 
     void createMissingShipmentItems(StockMovement stockMovement) {
-        Requisition requisition = stockMovement.requisition
+        Requisition requisition = stockMovement.requisition.refresh()
```
[the remark this answers is not in the document]
Comment (awalkowiak) [reply grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy:2369]: Please, check if the recall and revert recall refresh the product availability with this `refresh()` here?

[the remark this answers is not in the document]
Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy:2369]: I don't think it's missing, I think it was explicitly removed. Or maybe we used a Requisition.get() before. I think that would be fine.

**Review by jmiranda — CHANGES_REQUESTED** [review]

File: grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (drodzewicz) [reply grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy:2369]: Unfortunately reverting it back to `Requisition.get(stockMovement.id)` doesn't fix the issue with reviseItems.
I tested recall and product availability refresh with the `refresh()` and it seems to be working fine, everything seems to be updated immediately.
We could wait until tech huddle to discuss this case in more detail, but at this moment I don't have any other solution other than the one currently in the PR.

[the remark this answers is not in the document]
Comment (awalkowiak) [reply grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy:2372]: @drodzewicz I'm fine with restoring the `refresh()`, but could you add also a comment in the code on why it is needed there?

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**